### DISCUSSION TITLE
gauge run fatal error bsdthread_register error on MacOS Monterey

### DISCUSSION THREAD

**Discussion by haruki-kaneko** [getgauge/gauge#2211] [General]
Hi there

my environment is as follows:

```
❯ gauge -v
Gauge version: 1.4.3

Plugins
-------
html-report (4.0.12)
java (0.7.2)
screenshot (0.0.1)
spectacle (0.1.3)
xml-report (0.2.2)

❯ go version
go version go1.17.6 darwin/amd64
```

I'm seeing this error:

```
/Users/hoge/.asdf/shims/gauge run

Testing started at 21:38 ...
fatal error:
runtime: bsdthread_register error
runtime stack:
runtime.throw(0x129ac4c, 0x21)
	/usr/local/go/src/runtime/panic.go:616 +0x81 fp=0x7ff7bfefa618 sp=0x7ff7bfefa5f8 pc=0x10285c1
runtime.goenvs()
	/usr/local/go/src/runtime/os_darwin.go:129 +0x83 fp=0x7ff7bfefa648 sp=0x7ff7bfefa618 pc=0x1026143
runtime.schedinit()
	/usr/local/go/src/runtime/proc.go:
501 +0xd6 fp=0x7ff7bfefa6b0 sp=0x7ff7bfefa648 pc=0x102aeb6
runtime.rt0_go(0x7ff7bfefa6e0, 0x2, 0x7ff7bfefa6e0, 0x1000000, 0x2, 0x7ff7bfefa918, 0x7ff7bfefa927, 0x0, 0x7ff7bfefa92f, 0x7ff7bfefab7e, ...)
	/usr/local/go/src/runtime/asm_amd64.s:252 +0x1f4
fp=0x7ff7bfefa6b8 sp=0x7ff7bfefa6b0 pc=0x1050ff4
Error ----------------------------------
```

I've tried this on the following:

```
brew update
brew upgrade
```

I'd appreciate it if you could tell me what you know.

Thanks in advance!

**Comment by sriv** [comment]
This seems to be a problem that can happen when an older version of go is used to compile the program (see https://github.com/golang/go/issues/49425#issuecomment-964621714)

Could you please try the below options and see if it is any better?

1. use the pre-built binaries from github releases - https://github.com/getgauge/gauge/releases/download/v1.4.3/gauge-1.4.3-darwin.x86_64.zip
2. build from source - try `go install github.com/getgauge/gauge@v1.4.3` and run `~/go/bin/gauge`

**Reply by haruki-kaneko** [reply]
Thanks for the suggestion！

I tried both of them and got the same error....
